homeassistant/components/vicare/fan.py

```diff
@@ -137,6 +130,18 @@ def __init__(
 
     def update(self) -> None:
         """Update state of fan."""
+
+        # init presets
+        if self._attr_preset_modes is None:
+            supported_modes = list[str](self._api.getAvailableModes())
+            self._attr_preset_modes = [
+                mode
+                for mode in VentilationMode
+                if VentilationMode.to_vicare_mode(mode) in supported_modes
+            ]
+            if len(self._attr_preset_modes) > 0:
+                self._attr_supported_features |= FanEntityFeature.PRESET_MODE
```

Comment:
In another pr I was asked to put thinks that talk to the API to the update method, therefore I put it now here.

Reply:
Does the calls really do io or is the information already loaded and stored in the library so these calls only fetch it?

Reply:
It could, if the cache has expired.

Reply:
So data will be loaded upfront when `get_device_serial` is called. Therefore I moved the code to init. The revealed also that `get_device_serial` was not called in an separate executor.

Reply:
I didn't check in details but if it can do io I'd suggest you move it to `setup_entry` and do the necessary check there, to only directly set presets in `__init__`

Reply:
It will do no io as that has been done right upfront and the results are cached.